In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Chargement des données

In [30]:
# Chargement données
df = pd.read_csv('raw/retail_sales_dirty.csv')

# Exploration du Dataframe

## Informations Basiques

Format du dataframe, Colonnes, Types de Colonnes, Aperçu, Info

In [48]:
# Exploration du dataframe
print(f"Shape: {df.shape}")
print(f"\nColonnes:\n{df.columns.tolist()}")
print(f"\nTypes:\n{df.dtypes}")
print("\nInfo")
df.info()

Shape: (12575, 13)

Colonnes:
['Transaction ID', 'Customer ID', 'Category', 'Item', 'Price Per Unit', 'Quantity', 'Total Spent', 'Payment Method', 'Location', 'Transaction Date', 'Discount Applied', 'Calculated_Total', 'Diff']

Types:
Transaction ID       object
Customer ID          object
Category             object
Item                 object
Price Per Unit      float64
Quantity            float64
Total Spent         float64
Payment Method       object
Location             object
Transaction Date     object
Discount Applied     object
Calculated_Total    float64
Diff                float64
dtype: object

Info
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12575 entries, 0 to 12574
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Transaction ID    12575 non-null  object 
 1   Customer ID       12575 non-null  object 
 2   Category          12575 non-null  object 
 3   Item              11362 non-null

In [49]:

print("\nAperçu:")
#print(f"\nAperçu:\n{df.head(10)}")
df.head(10)


Aperçu:


Transaction ID Customer ID       Category          Item  Price Per Unit  \
0    TXN_6867343     CUST_09     Patisserie   Item_10_PAT            18.5   
1    TXN_3731986     CUST_22  Milk Products  Item_17_MILK            29.0   
2    TXN_9303719     CUST_02       Butchers   Item_12_BUT            21.5   
3    TXN_9458126     CUST_06      Beverages   Item_16_BEV            27.5   
4    TXN_4575373     CUST_05           Food   Item_6_FOOD            12.5   
5    TXN_7482416     CUST_09     Patisserie           NaN             NaN   
6    TXN_3652209     CUST_07           Food   Item_1_FOOD             5.0   
7    TXN_1372952     CUST_21      Furniture           NaN            33.5   
8    TXN_9728486     CUST_23      Furniture   Item_16_FUR            27.5   
9    TXN_2722661     CUST_25       Butchers   Item_22_BUT            36.5   

   Quantity  Total Spent  Payment Method  Location Transaction Date  \
0      10.0        185.0  Digital Wallet    Online       2024-04-08   
1       9.0        261.0  Digital Wallet    Online       2023-07-23   
2       2.0         43.0     Credit Card    Online       2022-10-05   
3       9.0        247.5     Credit Card    Online       2022-05-07   
4       7.0         87.5  Digital Wallet    Online       2022-10-02   
5      10.0        200.0     Credit Card    Online       2023-11-30   
6       8.0         40.0     Credit Card  In-store       2023-06-10   
7       NaN          NaN  Digital Wallet  In-store       2024-04-02   
8       1.0         27.5     Credit Card  In-store       2023-04-26   
9       3.0        109.5            Cash    Online       2024-03-14   

  Discount Applied  Calculated_Total  Diff  
0             True             185.0   0.0  
1             True             261.0   0.0  
2            False              43.0   0.0  
3              NaN             247.5   0.0  
4            False              87.5   0.0  
5              NaN               NaN   NaN  
6             True              40.0   0.0  
7             True               NaN   NaN  
8            False              27.5   0.0  
9            False             109.5   0.0

### Remarques
- 11 Colonnes, 12575 lignes
- Transaction Date et Discount Applied ont un type incorrect. Pour la correction à passer en date et booléen
- Valeurs manquantes dans plusieurs colonnes (Item, Price Per Unit, Quantity, Total Spent, Discount Applied). Pour la correction, Construire des règles pour compléter les valeurs manquantes ou les supprimer 

### Détails des Données Manquantes

In [37]:
df_Nul = pd.DataFrame({
    'Count': df.isnull().sum(),
    'Pourcentage': (df.isnull().sum() / df.shape[0] * 100).round(2)
    })
df_Nul=df_Nul[df_Nul["Count"]>0]
df_Nul

Count  Pourcentage
Item               1213         9.65
Price Per Unit      609         4.84
Quantity            604         4.80
Total Spent         604         4.80
Discount Applied   4199        33.39
Calculated_Total   1213         9.65
Diff               1213         9.65

### Doublons

In [56]:

# Doublons
print(f"Doublons d'entrées : {df.duplicated().sum()}")
#print(f"\nAperçu:\n{df.head(10)}")


print(f"\nUnicité des Transaction ID : {df['Transaction ID'].duplicated().sum()}")



Doublons d'entrées : 0

Unicité des Transaction ID : 0


Pas de doublons identifiés

### Cohérence
Vérification de la cohérence Price Per Unit x Quantity = Total Spent

In [62]:
df['Calculated_Total'] = df['Price Per Unit'] * df['Quantity']
df['Diff'] = (df['Total Spent'] - df['Calculated_Total']).abs()

Anomalie = df[df['Diff'] > 0.01].shape[0]

print(f"{Anomalie} Incohérence(s) détectée(s)")


0 Incohérence(s) détectée(s)
